### PR TITLE
transport socket: additional factory plumbing

### PR DESCRIPTION
**Pull request by mattklein123**
Required for the tap transport socket and I didn't realize it
in my previous PR.

*Risk Level*: Low
*Testing*: Existing tests
*Docs Changes*: N/A
*Release Notes*: N/A

### CODE REVIEW

File: include/envoy/upstream/cluster_manager.h [hunk 1]
```diff
@@ -315,6 +317,8 @@ class ClusterInfoFactory {
     const LocalInfo::LocalInfo& local_info_;
     Event::Dispatcher& dispatcher_;
     Runtime::RandomGenerator& random_;
+    Singleton::Manager& singleton_manager_;
+    ThreadLocal::SlotAllocator& tls_;
```
Comment (venilnoronha) [inline include/envoy/upstream/cluster_manager.h]: nit: s/tls/thread_local here and in the rest of the change?

Comment (mattklein123) [reply include/envoy/upstream/cluster_manager.h]: `thread_local` is a reserved word. I will leave it as is.

**Comment by mattklein123** [comment]
@lizan PTAL

File: include/envoy/upstream/cluster_manager.h (hunk 1 above)
Comment (venilnoronha) [reply include/envoy/upstream/cluster_manager.h]: Okay.

**Comment by mattklein123** [comment]
> In plumbing level this LGTM, a general question is how much more we want to do the plumbing like this, i.e. should we migrate some of them to be managed by singleton manager?

Yeah I'm not sure, that's a good question. Let me discuss this with @jmarantz. I think we need to start holistically figuring out if we want to just be passing around some type of `Api` object that holds most of these things.

**Pull request merged**